clients/packages/polarkit/src/api/errors.ts

```diff
@@ -0,0 +1,29 @@
+import { ApiError, ValidationError } from './client'
+
+interface APIValidationError extends ApiError {
+  body: {
+    detail: ValidationError[]
+  }
+}
+
+export const isValidationError = (error: ApiError): error is APIValidationError => {
+  return error.status === 422
+}
+
+type ValidationErrorsMap = Record<string, string[]>;
+
+export const getValidationErrorsMap = (errors: ValidationError[]): ValidationErrorsMap => {
+  return errors.reduce<ValidationErrorsMap>((map, error) => {
+    const loc = error.loc.slice(1).join('.')
+    if (map[loc]) {
+      return {
+        ...map,
+        [loc]: [...map[loc], error.msg],
+      }
+    }
+    return {
+      ...map,
+      [loc]: [error.msg],
+    }
+  }, {})
+}
```

Comment:
I had to add more muscles to our error handling in the UI. Basically, this helps us to work with FastAPI `ValidationError` coming from Pydantic.

Reply:
This is such a fantastic enhancement! It's been on the back of my mind for a bit, but never got around to it (a bit too happy path programming now during the early days). Love how simple this is and that with Pydantic we can easily add more user-friendly error messages for the future